### PR TITLE
Update TypeTestingAndConversionOperators.cs

### PR DESCRIPTION
**Pull request by Xamarin2019**
Fixed code snippet for array casting example.
image

Now it works in browser.
## Summary

Checking the code in the browser raised an error.

I'm happy to help Microsoft!

### CODE REVIEW

**Comment by Xamarin2019** [comment]
Dear team!
I think you need unit tests for code samples.
image

**Comment by BillWagner** [comment]
closing and reopening for a fresh CI build.

**Review by BillWagner — APPROVED** [review]
Thanks @Xamarin2019 

This LGTM, and I'll :shipit: as soon as the build finishes.

**Comment by BillWagner** [comment]
> Dear team! I think you need unit tests for code samples. 

We know. It's a harder problem than it appears at first. 

Most importantly, we don't want the introduction of unit tests to make the reader's experience worse. For example, it's easy to copy / paste / modify samples that use `Console.WriteLine` to experiment with an API or a concept. Those are hard to unit test, and brittle when you do. We're working on some technology that would enable us to "replay" the steps for code in an article and verify the results. Those show promise, but aren't perfect yet. 

In the meantime, we have a CI system to (at least) catch compiler errors.

File: docs/csharp/language-reference/operators/snippets/shared/TypeTestingAndConversionOperators.cs [hunk 1]
```diff
@@ -77,7 +77,7 @@ private static void IsOperatorDeclarationPattern()
     private static void AsOperator()
     {
         // <SnippetAsOperator>
-        IEnumerable<int> numbers = [10, 20, 30];
+        IEnumerable<int> numbers = {10, 20, 30};
```
Comment (BillWagner) [inline docs/csharp/language-reference/operators/snippets/shared/TypeTestingAndConversionOperators.cs:80]: Need to have a `new` expression here:

```suggestion
        IEnumerable<int> numbers = new List<int>(){10, 20, 30};
```

**Pull request merged**